docs/rfcs/0002-lambda-deployment.md

```diff
@@ -0,0 +1,87 @@
+- Start Date: 2020-12-21
+- Target Version: 1.0.0
+
+# Summary
+
+This RFC proposes adding a new service deployment from PipeCD: AWS Lambda deployment. Similar to the current [Google clould run deployment](https://pipecd.dev/docs/feature-status/#cloudrun-deployment).
+
+# Motivation
+
+PipeCD aims to support wide range of deployable services, currently [Terraform deployment](https://pipecd.dev/docs/feature-status/#terraform-deployment) and [CloudRun deployment](https://pipecd.dev/docs/feature-status/#cloudrun-deployment) are supported. Lambda deployment is the current missing piece for PipeCD's purpose.
+
+# Detailed design
+
+Note:
+- Since the `Lambda function as container image` feature was available on AWS recently, we plan to implement PipeCD Lambda deployment based on that update.
+- To simplify the implementation, the initial release of this feature will only support container images from ECR ( the AWS container registry ).
+
+### Usage
+
+The deployment configuration is used to customize the way to do the deployment. In case of Lambda function deployment, current common stage options (`WAIT`, `WAIT_APPROVAL`, `ANALYSIS`) are all inherited, besides with the stages for Lambda deployment itself: `LAMBDA_CANARY_ROLLOUT`, `LAMBDA_TRAFFIC_ROUTING`.
+
+```yaml
+# https://docs.aws.amazon.com/lambda/latest/dg/configuration-versions.html
+apiVersion: pipecd.dev/v1beta1
+kind: LambdaApp
+spec:
+  input:
+    image: xxxxxxxxxxxx.dkr.ecr.ap-northeast-1.amazonaws.com/demoapp:v0.0.1
```

Comment:
Are there any other fields that can be configured for Lambda?
With CloudRun we can configure the concurrency number, the memory and CPU limit, etc...

Reply:
As far as I investigated, we could set the memory and CPU limit with the same pattern we did in case of CloudRun. Need to investigate more for concurrency setting 🙏 ( that feature could be caught on lambda UI, so I though we could basically )

Reply:
Got it. I just want to say that because we are following GitOps pattern so it would be nice if all configurations can be committed and managed via Git. But they can be added in the future. 👍

Reply:
got it, thank you 🙏